### PR TITLE
Allow an explicit mapping between tprops and scalars for contour labels.

### PR DESCRIPTION
**Pull request by alliepiper**
Goes with UV-CDAT/uvcdat#1203

### CODE REVIEW

File: Rendering/Core/Testing/Cxx/TestLabeledContourMapper.cxx [hunk 1]
```diff
@@ -74,9 +77,33 @@ int TestLabeledContourMapper(int argc, char *argv[])
   tprop3->SetColor(.3, .8, .2);
   tprops->AddItem(tprop3.GetPointer());
```
Comment (aashish24) [inline Rendering/Core/Testing/Cxx/TestLabeledContourMapper.cxx:79]: Why only 6?

Comment (alliepiper) [inline Rendering/Core/Testing/Cxx/TestLabeledContourMapper.cxx:79]: There are places in the mapper code that handle the case where numTProps < numContours. This exercises that functionality.

**Comment by aashish24** [comment]
:+1:

**Comment by alliepiper** [comment]
@aashish24 Can we get this into upstream as well? https://gitlab.kitware.com/vtk/vtk/merge_requests/144

**Comment by doutriaux1** [comment]
@dlonie @aashish24 is there anyway that all of our "custom" VTK PRs also make it into our "Custom" Paraview/VTK for when we use BUILD_MODE=ALL

**Comment by aashish24** [comment]
ParaView/VTK would be risky since we won't know if the changes are compatible.

**Comment by doutriaux1** [comment]
@dlonie what do you think? Can we try it and run the paraview test suite?

**Comment by alliepiper** [comment]
I'm honestly not sure why we're maintaining forks of both. Why not just use one or the other? 

If both are needed, it would help to keep both closer to upstream master. That would make it easier to maintain any divergence/patches.

I can go ahead and say that if we put our VTK into the current ParaView branch, we'll have compilation errors. The two projects are very tightly synced and ParaView is frequently updated to reflect changes to VTK's master.

**Comment by doutriaux1** [comment]
I kind of agree. My understanding is that we usually do not want ParaView because it is too big.

Also it seems that ParaView anyway clones VTK, and I'm not positive that our ParaView clones our VTK fork. I think it might be the best solution have our ParaView fork clone our VTK fork at uvcdat-master? Is that hard to do?

**Comment by alliepiper** [comment]
It could be done, but it would be a bit of work and maintenance effort to keep them in sync. If there's some way to just use one or the other (preferably just VTK, as we wouldn't need to deal with syncing submodules...), that would be much, much better.

What do we need from ParaView that VTK doesn't provide?

**Comment by doutriaux1** [comment]
right now nothing. but I think @aashish24 wants to use for some parallel things also I think @jbeezley and @mattben need ParaView for their web uvcdat.

**Comment by alliepiper** [comment]
Sounds like this could use some clarification/discussion. Should we move this to the mailing list?

**Comment by jbeezley** [comment]
For the record, we use VTK for cdatweb.

**Comment by doutriaux1** [comment]
oh good! Thanks for the clarification @jbeezley then we should wait to hear from @aashish24 but it sounds like we can probably turn off ParaView for now. (although I really like the GUI for quick inspections)

**Pull request merged**